### DISCUSSION TITLE
Splitting loaders and default exports for a route for Vite

### DISCUSSION THREAD

**Discussion by meza** [remix-run/remix#9323] [Q&A] [answered]
Hey,

I'm in the process of migrating over to Vite and one of the problems I'm facing is that some of my loaders are most definitely server only (as intended by Remix) but now Vite is complaining. 

Can I split the loaders out to a different .server file? If so, how would I configure the routes? (I use custom config for the routes)

**Comment by sergiodxa** [comment, marked as answer]
You can move them to a `.server` file and then re-export them. Let's say you have the route `routes/something.tsx` with a loader and component, you can rename it to `routes/something/route.tsx` then create a `routes/something/loader.server.ts` and in the route.tsx file add:

```ts
export { loader } from "./loader.server";
```

**Reply by meza** [reply]
Thanks! This feels like it shouldn't work but it does!
